### PR TITLE
Feature/target details

### PR DESCRIPTION
**Pull request by jpirazusta**
User can see target details after selecting one on the map.
gif

### CODE REVIEW

**Review by ManuViola77 — APPROVED** [review]
Nicely done!! 👏

File: src/components/CustomMarker/index.js [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
 import { useSelector } from 'react-redux';
-import { number } from 'prop-types';
+import { number, func, string, bool } from 'prop-types';
 import { Marker } from 'react-native-maps';
 import _ from 'lodash';
 
 import styles from './styles';
 
-const CustomMarker = ({ lat, lng, topicId }) => {
+const CustomMarker = ({ id, lat, lng, topicId, onPress, selected }) => {
```
Comment (peter7z) [inline src/components/CustomMarker/index.js]: This component's name makes me think I could customize the icon or sth. Maybe TargetMarker could be better?

File: src/constants/colors.js [hunk 2]
```diff
@@ -6,3 +6,5 @@ export const TRANSPARENT = 'rgba(255, 255, 255, 0)';
 export const MARKER = 'rgba(239, 198, 56, 0.7)';
 export const MARKER_BORDER = '#EFC638';
 export const MARKER_CIRCLE = '#rgba(255, 255, 255, 0.7)';
+export const SELECTED_TARGET = 'rgba(47, 188, 247, 0.7)';
```
Comment (peter7z) [inline src/constants/colors.js]: I prefer color names for color constants.

File: src/components/TargetForm/TargetFormContent.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  );
+
+  const {
+    handleOnPress,
+    targetError,
+    targetStatus,
+    getTopicsError,
+    getTopicsStatus,
+    topicError,
+  } = useCreateTargetForm(selectedTopic, emptyTopic, handleSubmit);
```
Comment (peter7z) [inline src/components/TargetForm/TargetFormContent.js]: Maybe you could

File: src/components/TargetForm/TargetFormContent.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  useEffect(() => {
+    if (existent) {
+      const {
+        topic: { label },
+      } = _.find(topics, ({ topic: { id } }) => id === existent.topicId);
+      setActualTopic(label);
+    }
+  }, [existent, setActualTopic, topics]);
+
+  const { values, errors, handleValueChange, handleSubmit, handleBlur } = useForm(
```
Comment (peter7z) [inline src/components/TargetForm/TargetFormContent.js]: Maybe you could move this hook call into the useTargetForm hook?

File: src/screens/MainScreen/index.js [hunk 5]
```diff
@@ -41,20 +48,49 @@ const MainScreen = () => {
...
+  }, [location, selectedTarget]);
+
+  const onSelectTarget = useCallback(
+    (id, targets) => {
+      const { target } = _.find(targets, ({ target }) => target.id === id);
+      setSelectedTarget(target);
+      animate(formPositionAnim, 0);
+    },
+    [formPositionAnim],
+  );
```
Comment (peter7z) [inline src/screens/MainScreen/index.js]: Maybe you could move these into a hooks file, specific to this screen/component.
Custom hooks are a great way to encapsulate logic, be it for later reuse or just moving it into another file to declutter a component.

File: src/components/common/Button/index.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ const Button = ({ testID, handleOnPress, additionalStyles, title, isLoading }) =
 Button.propTypes = {
   testID: string,
   handleOnPress: func.isRequired,
-  additionalStyles: object.isRequired,
+  additionalStyles: oneOfType([arrayOf(object), object]),
```
Comment (peter7z) [inline src/components/common/Button/index.js]:
```suggestion
  additionalStyles: oneOfType([arrayOf(ViewPropTypes.style), ViewPropTypes.style]),
```

You can import `ViewPropTypes` from `reat-native`.
I think that just this should work too, not sure though.
```suggestion
  additionalStyles: ViewPropTypes.style,
```

**Review by peter7z — APPROVED** [review]
Very nice!

File: src/constants/colors.js [hunk 7]
```diff
@@ -1,8 +1,10 @@
...
-export const ERROR = '#E02424';
+export const RED_ERROR = '#E02424';
 export const BLACK = '#000000';
 export const TRANSPARENT = 'rgba(255, 255, 255, 0)';
-export const MARKER = 'rgba(239, 198, 56, 0.7)';
-export const MARKER_BORDER = '#EFC638';
-export const MARKER_CIRCLE = '#rgba(255, 255, 255, 0.7)';
+export const YELLOW_MARKER = 'rgba(239, 198, 56, 0.7)';
+export const YELLOW = '#EFC638';
+export const LIGHT_YELLOW = '#rgba(255, 255, 255, 0.7)';
```
Comment (elizabethlofredo) [inline src/constants/colors.js]:
```suggestion
export const LIGHT_YELLOW = 'rgba(255, 255, 255, 0.7)';
```

File: src/hooks/useSelectTarget.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { useEffect, useCallback } from 'react';
+import _ from 'lodash';
```
Comment (elizabethlofredo) [inline src/hooks/useSelectTarget.js]: Here you import all lodash but you are using just `find` function, you can do:

```suggestion
import find from 'lodash/find';
```
and use find directly

File: src/hooks/useTargetForm.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import { useEffect } from 'react';
+import _ from 'lodash';
```
Comment (elizabethlofredo) [inline src/hooks/useTargetForm.js]: same here

File: src/screens/MainScreen/index.js [hunk 10]
```diff
@@ -41,20 +48,39 @@ const MainScreen = () => {
...
-          longitudeDelta: LONGITUDE_DELTA,
+          ...coordsConstants,
         }}
-        showUserLocation>
+        showUserLocation
+        followUserLocation>
         {topics &&
-          targets.map(({ target: { id, lat, lng, topicId } }) => (
-            <CustomMarker key={id} lat={lat} lng={lng} topicId={topicId} />
+          targets.map(({ target: { id, lat, lng, topicId } }, _, targets) => (
```
Comment (elizabethlofredo) [inline src/screens/MainScreen/index.js]:
```suggestion
          targets.map(({ target: { id, lat, lng, topicId } }) => (
```
You don't need to get the array from the map, you already have it

File: src/screens/MainScreen/index.js [hunk 11]
```diff
@@ -41,20 +48,39 @@ const MainScreen = () => {
...
+            <TargetMarker
+              key={id}
+              id={id.toString()}
+              lat={lat}
+              lng={lng}
+              topicId={topicId}
+              onPress={() => onSelectTarget(id, targets)}
+              selected={selectedTarget && selectedTarget.id === id}
```
Comment (elizabethlofredo) [inline src/screens/MainScreen/index.js]: Here you also can do:
```suggestion
              selected={selectedTarget?.id === id}
```

**Review by elizabethlofredo — APPROVED** [review]
Great job!! I left some comments 😄

File: src/constants/colors.js (hunk 7 above)
Comment (jpirazusta) [reply src/constants/colors.js]: Oops! Thanks!

**Pull request merged**